src/pretix/plugins/socialauth/views.py

```diff
@@ -18,43 +20,107 @@
 from pretix.helpers.urls import build_absolute_uri
 
 from .schemas.login_providers import LoginProviders
+from .schemas.oauth2_params import OAuth2Params
 
 logger = logging.getLogger(__name__)
 adapter = get_adapter()
 
 
-def oauth_login(request, provider):
-    gs = GlobalSettingsObject()
-    client_id = gs.settings.get('login_providers', as_type=dict).get(provider, {}).get('client_id')
-    provider = adapter.get_provider(request, provider, client_id=client_id)
-
-    base_url = provider.get_login_url(request)
-    query_params = {
-        "next": build_absolute_uri("plugins:socialauth:social.oauth.return")
-    }
-    parsed_url = urlparse(base_url)
-    updated_url = parsed_url._replace(query=urlencode(query_params))
-    return redirect(urlunparse(updated_url))
+class OAuthLoginView(View):
+    def get(self, request: HttpRequest, provider: str) -> HttpResponse:
+        self.set_oauth2_params(request)
 
+        gs = GlobalSettingsObject()
+        client_id = (
+            gs.settings.get("login_providers", as_type=dict)
+            .get(provider, {})
+            .get("client_id")
+        )
+        provider_instance = adapter.get_provider(request, provider, client_id=client_id)
+
+        base_url = provider_instance.get_login_url(request)
+        query_params = {
+            "next": build_absolute_uri("plugins:socialauth:social.oauth.return")
+        }
+        parsed_url = urlparse(base_url)
+        updated_url = parsed_url._replace(query=urlencode(query_params))
+        return redirect(urlunparse(updated_url))
+
+    @staticmethod
+    def set_oauth2_params(request: HttpRequest) -> None:
+        """
+        Handle Login with SSO button from other components
+        This function will set 'oauth2_params' in session for oauth2_callback
+        """
+        next_url = request.GET.get("next", "")
+        if not next_url:
+            return
+
+        parsed = urlparse(next_url)
+
+        # Only allow relative URLs
+        if parsed.netloc or parsed.scheme:
+            return
+
+        params = parse_qs(parsed.query)
+        sanitized_params = {
```

Comment:
You don't need this step if you have Pydantic model. You can tell Pydantic to automatically strip whitespace.

```py
from typing import Annotated

from pydantic import BaseModel, StringConstraints


class OAuth2Params(BaseModel):
     response_type: Annotated[str, StringConstraints(strip_whitespace=True)] = 'code'
```

Learn more [here](https://docs.pydantic.dev/latest/api/types/#pydantic.types.conbytes).